src/source.rs

```diff
@@ -11,6 +11,7 @@ use ffi::{gboolean, gpointer};
 use translate::{from_glib, FromGlib, ToGlib};
 
 /// The id of a source that is returned by `idle_add` and `timeout_add`.
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq)]
```

Comment:
it is usefull. A default value of zero is good because it is an invalid source and used that way in glib.